src/dbnode/namespace/schema.go

```diff
@@ -251,8 +256,22 @@ func genDependencyDescriptors(infd *desc.FileDescriptor) []*desc.FileDescriptor
 	return depfds
 }
 
-func parseProto(protoFile string, importPaths ...string) ([]*desc.FileDescriptor, error) {
-	p := protoparse.Parser{ImportPaths: importPaths, IncludeSourceCodeInfo: true}
+// protoStringProvider provides proto contents from strings.
+func protoStringProvider(source map[string]string) protoparse.FileAccessor {
+	if len(source) == 0 {
```

Comment:
should it just return an optional error? Cant really do anything with a nil FileAccessor

Reply:
sure, originally, nil FileAccessor meant Parser will use os.Open to source proto from local file system. 
we can be more explicitly about this.

Reply:
actually, I don't see anything wrong with return a nil FileAccessor, protoparse.Parser take a nil FileAccessor, if map is empty, it is the right response to return a nil FileAccessor.

Reply:
Ok sounds good if thats what the Parser package expects. Maybe just a comment saying how that works